paimon-core/src/main/java/org/apache/paimon/operation/RawFileSplitRead.java

```diff
@@ -212,14 +214,20 @@ private RecordReader<InternalRow> createFileReader(
                         dataFilePathFactory.toPath(file.fileName()),
                         file.fileSize(),
                         fileIndexResult);
-        FileRecordReader fileRecordReader =
+        RecordReader<InternalRow> fileRecordReader =
                 new FileRecordReader(
                         bulkFormatMapping.getReaderFactory(),
                         formatReaderContext,
                         bulkFormatMapping.getIndexMapping(),
                         bulkFormatMapping.getCastMapping(),
                         PartitionUtils.create(bulkFormatMapping.getPartitionPair(), partition));
 
+        if (fileIndexResult instanceof BitmapIndexResult) {
```

Comment:
Here should be bug, bitmap in `ApplyBitmapIndexRecordReader` should be merged with `ApplyDeletionVectorReader`, otherwise the latter will encounter problems.

Can you create a test for this?

Reply:
Could you be more specific for the bug?

As I understand it
The bitmap index reader would filter all the mismatched data
The DV reader would filter all the deleted data
Is there a conflict?

Can this test case be covered?
org.apache.paimon.table.PrimaryKeyFileStoreTableTest#testDeletionVectorsWithFileIndexInMeta
org.apache.paimon.table.PrimaryKeyFileStoreTableTest#testDeletionVectorsWithBitmapFileIndexInFile

Reply:
You are right, the magic is in the returned position, it makes chain processing possible.